packages/core/src/tracing/trace.ts

```diff
@@ -399,7 +399,8 @@ type SpanWithScopes = Span & {
   [ISOLATION_SCOPE_ON_START_SPAN_FIELD]?: Scope;
 };
 
-function setCapturedScopesOnSpan(span: Span | undefined, scope: Scope, isolationScope: Scope): void {
+/** Store the scope & isolation scope for a span, which can the be used when it is finished. */
+export function setCapturedScopesOnSpan(span: Span | undefined, scope: Scope, isolationScope: Scope): void {
```

Comment:
we need this in otel!

Reply:
h: We should just vendor this in the otel package. This function should not be exposed to the user.

Reply:
not sure, this is subject to break if we do that as that is not typed anywhere properly and relies purely on casting stuff around etc 😬 we _need_ this to be the exact same implementation in node-experimental as here, as we _need_ this to be picked up by `getCaputredScopesOnSpan()`.